### PR TITLE
[FIX] mass_mailing: remove inactive lists from mass mailing

### PR DESCRIPTION
**Pull request by chienandalu**
Backport from #46622

Issue

	- Install Mass Mailing
	- Create 2 mailing list (x & y) with differents
	  contacts
	- Create a mass mailing with x & y, send it

	Ok, sent to everyome

	- Archive mailing list y
	- Duplicate the last mass mailing, send it

	Not ok, sent to y contacts too

	When a list is archived, I can't manually
	remove it from the mass mailing (it disappears)

Cause

	When archiving a list, it still remains in
	contact_list_ids

Solution

	Prevent archiving a mailing list if any mass mailing
	using it exist with state != 'done'

	Re-evaluate the domain field at copy

OPW-2202111

closes odoo/odoo#46622

Signed-off-by: Nicolas Martinelli (nim) <nim@odoo.com>

cc @Tecnativa TT22469

### CODE REVIEW

**Review by rafaelbn — APPROVED** [review]

File: addons/mass_mailing/models/mass_mailing.py [hunk 1]
```diff
@@ -433,9 +447,16 @@ def name_create(self, name):
...
+        # Cleaning archived contact_list_ids
         default = dict(default or {},
-                       name=_('%s (copy)') % self.name)
-        return super(MassMailing, self).copy(default=default)
+                       name=_('%s (copy)') % self.name,
+                       contact_list_ids=self.contact_list_ids.ids)
+        res = super(MassMailing, self).copy(default=default)
+        # Re-evaluating the domain
+        body_html = res.body_html
+        res._onchange_model_and_list()
```
Comment (pedrobaeza) [inline addons/mass_mailing/models/mass_mailing.py:457]: Does this work the same way in this version?

Comment (chienandalu) [reply addons/mass_mailing/models/mass_mailing.py:457]: Yes. I just tested it on runbot and it works as expected

**Review by pedrobaeza — APPROVED** [review]

File: addons/mass_mailing/models/mass_mailing.py [hunk 2]
```diff
@@ -40,6 +40,20 @@ def _compute_contact_nbr(self):
         mapped_data = dict([(c['list_id'][0], c['list_id_count']) for c in contacts_data])
         for mailing_list in self:
             mailing_list.contact_nbr = mapped_data.get(mailing_list.id, 0)
+    
+    @api.multi
+    def write(self, vals):
+        # Prevent archiving used mailing list
+        if 'active' in vals and not vals.get('active'):
```
Comment (Tardo) [inline addons/mass_mailing/models/mass_mailing.py:47]: `get` check for you if the key exists in the dictionary ;) (if not, return None or the value that you given)
```suggestion
        if vals.get('active') is False:
```

**Review by Tardo — CHANGES_REQUESTED** [review]
Looks good! only one comment...

File: addons/mass_mailing/models/mass_mailing.py (hunk 2 above)
Comment (pedrobaeza) [reply addons/mass_mailing/models/mass_mailing.py:47]: It's not the same, as you want to check that you are writing `active=False`, so no way to distinguish that from an nonexistent value in the dictionary.

Comment (Tardo) [reply addons/mass_mailing/models/mass_mailing.py:47]: Updated... if 'get' returns 'False' is because 'vals' have the key 'active' and is False.

Comment (pedrobaeza) [reply addons/mass_mailing/models/mass_mailing.py:47]: Well, the performance is the same, but the most important thing is that this is a backport, got as is from the original committer, so better to not change anything for seeing the same when comparing between versions. Next time, discuss this in the upstream PR :stuck_out_tongue_closed_eyes:

**Review by Tardo — APPROVED** [review]

**Pull request merged**